tests/integration/bank/keeper/deterministic_test.go

```diff
@@ -212,51 +197,49 @@ func (suite *DeterministicTestSuite) TestGRPCQuerySpendableBalances() {
 			coins = sdk.NewCoins(append(coins, coin)...)
 		}
 
-		suite.mockFundAccount(addr)
 		err := banktestutil.FundAccount(suite.bankKeeper, suite.ctx, addr, coins)
 		suite.Require().NoError(err)
 
-		suite.runSpendableBalancesIterations(addr, coins)
+		req := banktypes.NewQuerySpendableBalancesRequest(addr, testdata.PaginationGenerator(t, uint64(numCoins)).Draw(t, "pagination"))
+		res, err := suite.queryClient.SpendableBalances(suite.ctx, req)
+		suite.Require().NoError(err)
+		suite.runSpendableBalancesIterations(req, res)
 	})
 
-	suite.SetupTest() // reset
-	addr, err := sdk.AccAddressFromBech32("cosmos139f7kncmglres2nf3h4hc4tade85ekfr8sulz5")
-	suite.Require().NoError(err)
-
 	coins := sdk.NewCoins(
 		sdk.NewCoin("stake", sdk.NewInt(10)),
 		sdk.NewCoin("denom", sdk.NewInt(100)),
 	)
 
-	suite.mockFundAccount(addr)
-	err = banktestutil.FundAccount(suite.bankKeeper, suite.ctx, addr, coins)
+	err := banktestutil.FundAccount(suite.bankKeeper, suite.ctx, addr1, coins)
 	suite.Require().NoError(err)
 
-	suite.runSpendableBalancesIterations(addr, coins)
+	suite.runSpendableBalancesIterations(
+		banktypes.NewQuerySpendableBalancesRequest(addr1, nil),
+		&banktypes.QuerySpendableBalancesResponse{Balances: coins, Pagination: &query.PageResponse{Total: 2}},
+	)
 }
 
-func (suite *DeterministicTestSuite) runTotalSupplyIterations(prevRes sdk.Coins) {
-	for i := 0; i < 1000; i++ {
-		res, err := suite.queryClient.TotalSupply(suite.ctx, &banktypes.QueryTotalSupplyRequest{})
+func (suite *DeterministicTestSuite) runTotalSupplyIterations(req *banktypes.QueryTotalSupplyRequest, prevRes *banktypes.QueryTotalSupplyResponse) {
+	for i := 0; i < iterCount; i++ {
+		res, err := suite.queryClient.TotalSupply(suite.ctx, req)
 
 		suite.Require().NoError(err)
 		suite.Require().NotNil(res)
 
-		suite.Require().Equal(res.GetSupply(), prevRes)
-		prevRes = res.GetSupply()
+		suite.Require().Equal(res, prevRes)
 	}
 }
 
 func (suite *DeterministicTestSuite) TestGRPCQueryTotalSupply() {
-	rapid.Check(suite.T(), func(t *rapid.T) {
-		suite.SetupTest() // reset
+	suite.SetupTest() // reset
```

Comment:
https://github.com/cosmos/cosmos-sdk/pull/13450/files#r988741664

Reply:
Fixed in b233ab4743b436aeb5c60ba5f8941f84d3f77295

Reply:
I also fixed TestGRPCSendEnabled in ab8bcee